src/LLVM_Runtime_Linker.cpp

```diff
@@ -376,18 +372,19 @@ void link_modules(std::vector<llvm::Module *> &modules, Target t) {
     for (size_t i = 1; i < modules.size(); i++) {
         string err_msg;
         #if LLVM_VERSION >= 38
-        bool failed = llvm::Linker::linkModules(*modules[0], std::unique_ptr<llvm::Module>(modules[i]));
-        #elif LLVM_VERSION >= 36
-        bool failed = llvm::Linker::LinkModules(modules[0], modules[i]);
+        bool failed = llvm::Linker::linkModules(*modules[0],
+                                                std::move(modules[i]));
         #else
-        bool failed = llvm::Linker::LinkModules(modules[0], modules[i],
+            #if LLVM_VERSION >= 36
+            bool failed = llvm::Linker::LinkModules(modules[0], modules[i]);
+            #else
+            bool failed = llvm::Linker::LinkModules(modules[0], modules[i],
                                                 llvm::Linker::DestroySource, &err_msg);
-        #endif
+            #endif
 
-        // No matter which version about ran, we passed ownership of
-        // the second module to LLVM. modules[i] is now a dangling
-        // pointer. null it out so we don't accidentally use it.
-        modules[i] = nullptr;
+            // LLVM consumed the module so reset the unique_ptr.
+            modules[i].reset();
```

Comment:
Actually the code here is reseting it because prior to the latest LLVM, the LinkModules call took llvm::Module *, not std_unique_ptr<llvm::Module>. Of course there were some missing gets up above this as well. The joys of ifdefy code.